### DISCUSSION TITLE
The future of Mantine DataTable

### DISCUSSION THREAD

**Discussion by icflorescu** [icflorescu/mantine-datatable#450] [Announcements]
Dear community members,

As most of you already know, Mantine v7 came up with [a significant number of breaking changes, most of them referring to how styling is implemented](https://mantine.dev/changelog/7-0-0/).

As some of you know, `mantine-datatable` is one of my dearest open-source projects, has quite a large code-base, and while it has nearly 700 GitHub stars and 40k npm downloads per month as of November 2nd 2023, I'm still [struggling with raising enough funds](https://github.com/sponsors/icflorescu) to be able to dedicate as much time as I'd like to the project, despite the fact that it's [being used by lots of startups and developers](https://icflorescu.github.io/mantine-datatable/who-is-using-mantine-datatable) worldwide.

A lot of you [were looking forward](https://github.com/icflorescu/mantine-datatable/issues/420) for the new version supporting Mantine V7, which, after more than a month of assiduous work,  finally [came out yesterday](https://github.com/icflorescu/mantine-datatable/discussions/445), partly due to the generosity of some members who were kind enough to contribute financially towards the goal.

I must, however, point out the fact that - according to my calculations - less than 25% of the effort behind the upgrade process was financially compensated one way or the other. That's without even taking into consideration the entire 13 months long project lifespan prior to that.

I've been around a while, so I know that all open-source creators face similar issues. I also know it from my own experience. Mantine DataTable is one of my dearest open-source projects, but not the only one. I'm also the guy behind [Mantine ContextMenu](https://icflorescu.github.io/mantine-contextmenu/), [tRPC-SvelteKit](https://icflorescu.github.io/trpc-sveltekit), [PocketBaseUML](https://pocketbase-uml.github.io/), [Expose-WSL](https://github.com/icflorescu/expose-wsl), [IISExpress-Proxy](https://github.com/icflorescu/iisexpress-proxy) and [a few others](https://github.com/icflorescu), not to mention the contributions I'm constantly making to other people's or organizations' repos. While I don't do it primarily for money, I need money to support my existence, as everyone else does.

Please understand that this is not a rant - I'm simply pointing the facts here. I'll do my best to keep maintaining the package, come up with new features and fixing bugs as they emerge, but I can't do it without your support.

**I need [your help](https://github.com/sponsors/icflorescu).
A couple of companies backing up the project on a monthly basis, or one company that uses the component and hires [my services](https://github.com/icflorescu) would do wonders for future of the project on the longer term.**

Code contributions are also welcome, and hopefully [contributing](https://icflorescu.github.io/mantine-datatable/contribute-and-support/#become-a-code-contributor) should be easier starting with V7, following the major repo overhaul.

Thank you all for your patience and support!
Huge thanks to @john-latham for being the major financial contributor to support the V7 release 🙏❤️🙏
Special thanks to @KevinVandy, @hughfitzgerald, @simeq, @onlyice, @grahampcharles, @DoubleJ-G, @otobot1, @Mikkelzu, @blentz100, @beeman, @ericschmar, @aniravi24, @JitPackJoyride, @johnzanussi, @rakutek, @petrvecera, @usequantumai, @israelins85, @ecatugy, @zeitderforschung, @Emil-Sorensen, @stefanomarra, @AlexcastroDev, @samueldobbie and other people who sponsored me in the past, plan to sponsor me in the future, or contributed code to the repository (apologies if I omitted anyone). 

[![Mantine DataTable V7.1](https://github.com/icflorescu/mantine-datatable/assets/581999/72111cde-0404-467c-a333-aa68129b1a24)](https://icflorescu.github.io/mantine-datatable/)

**Comment by Thanaen** [comment]
Hello !
What would you think of using the [polar.sh](https://polar.sh/) platform to facilitate this kind of funding, in addition to the current Github sponsors?

**Reply by icflorescu** [reply]
An interesting suggestion. Some people also mentioned Patreon as an alternative.
May I ask why do you think it would be better that the current GitHub sponsors program?
Are there, to your knowledge, any hidden, non-obvious things that would prevent somebody (or a company) to fund through GitHub, but incentivize them to do so through  Polar or Patreon?

I thought GitHub sponsors was the "de-facto" standard, and best suited, since this is where the source code is hosted...
But maybe you're right, and offering more than just one option would be better.

Ah, well... another thing to add to the endless todo list 😮‍💨😊

**Reply by Thanaen** [reply]
I don't think it's better than Github Sponsors, I think it's complementary!
Github Sponsors are great for monthly and one-off donations. Polar.sh adds the ability for users to donate to a specific issue, so if a user has a really big issue, they can fund its resolution 😁

**Reply by Thanaen** [reply]
(To be honest, I've never used polar.sh myself, but I noticed that @zoontek was in a similar situation to you, and adopted Polar.sh for his [react-native-permissions](https://github.com/zoontek/react-native-permissions) project)

**Reply by icflorescu** [reply]
I see. Thanks a lot for the heads up and for taking the time to explain!

**Comment by icflorescu** [comment]
[Mantine DataTable v8.x](https://icflorescu.github.io/mantine-datatable/) released! :partying_face:
